realm/realm-library/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -794,4 +808,13 @@ public String toString() {
     public ProxyState realmGet$proxyState() {
         return proxyState;
     }
+
+    // Checks if the given field is primary key field. Throws if it is a PK field.
+    private void checkIsPrimaryKey(String fieldName) {
+        RealmObjectSchema objectSchema = proxyState.getRealm$realm().getSchema().getSchemaForClass(getType());
+        if (objectSchema.hasPrimaryKey() && objectSchema.getPrimaryKey().equals(fieldName)) {
+            throw new RealmException(String.format("Primary key field '%s' cannot be changed after object created."
```

Comment:
This is only for DynamicRealmObject, so it is called from e.g `setInt("myId", 1)`, right? So the IllegalArgument is for the the "myId" argument.